agora/Agora/Stake/Scripts.hs

```diff
@@ -154,30 +156,35 @@ stakePolicy gtClassRef =
     mintedST <- pletC $ psymbolValueOf # ownSymbol # txInfoF.mint
 
     let burning = unTermCont $ do
-          pguardC "ST at inputs must be 1" $
-            spentST #== 1
+          let numStakeInputs =
+                pto $
+                  pfoldMap @_ @_ @(PSum PInteger)
+                    # plam
+                      ( \((pfield @"resolved" #) -> txOut) -> unTermCont $ do
+                          txOutF <- pletFieldsC @'["value", "datum"] txOut
+
+                          let isStakeUTxO =
+                                psymbolValueOf # ownSymbol # txOutF.value #== 1
+
+                          pmatchC isStakeUTxO
+                            >>= \case
+                              PTrue -> do
+                                let datum =
+                                      pfromData $
+                                        pfromOutputDatum @(PAsData PStakeDatum)
+                                          # txOutF.datum
+                                          # txInfoF.datums
+
+                                pguardC "Stake is unlocked" $
+                                  pnot # (pstakeLocked # datum)
+
+                                pure $ pcon $ PSum 1
+                              PFalse -> pure mempty
+                      )
+                    # pfromData txInfoF.inputs
 
           pguardC "ST burned" $
-            mintedST #== -1
-
-          pguardC "An unlocked input existed containing an ST" $
-            pany
-              # plam
-                ( \((pfield @"resolved" #) -> txOut) -> unTermCont $ do
-                    txOutF <- pletFieldsC @'["value", "datum"] txOut
-                    pure $
-                      pif
-                        (psymbolValueOf # ownSymbol # txOutF.value #== 1)
-                        ( let datum =
-                                pfromData $
-                                  pfromOutputDatum @(PAsData PStakeDatum)
-                                    # txOutF.datum
-                                    # txInfoF.datums
-                           in pnot # (pstakeLocked # datum)
-                        )
-                        (pconstant False)
-                )
-              # pfromData txInfoF.inputs
+            mintedST #== numStakeInputs
```

Comment:
Shouldn't you negate the `numStakeInputs`?

Reply:
Yeah, I should. It's curious and quite concerning that the test suite didn't catch it though.